hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java

```diff
@@ -503,6 +503,11 @@ private void fillInbox() {
         if (instreamCursor == null) {
             return;
         }
+        // We need to collect metrics before draining the queues into Inbox,
+        // otherwise they would appear empty even for slow processors
+        queuesCapacity.set(sum(instreamCursor.getList(), InboundEdgeStream::capacities));
+        queuesSize.set(sum(instreamCursor.getList(), InboundEdgeStream::sizes));
+
```

Comment:
Why is this not before the null check above to report 0 in the metrics?

Reply:
that is a good question. I don't know under what circumstances the cursor can be null.

Reply:
hm, I guess it makes sense. @viliam-durina wdyt?

Reply:
It's been a while since I understood that code :)
`instreamCursor` is null when all input queues have completed. So it makes sense to set 0 to the metrics.

Reply:
ok, I just pushed an update